test/OpenTelemetry.Instrumentation.AspNetCore.Tests/MetricTests.cs

```diff
@@ -0,0 +1,136 @@
+        }
+
+        [Fact]
+        public async Task RequestMetricIsCaptured()
+        {
+            var metricItems = new List<MetricItem>();
+            var metricExporter = new TestExporter<MetricItem>(ProcessExport);
+
+            void ProcessExport(Batch<MetricItem> batch)
+            {
+                foreach (var metricItem in batch)
+                {
+                    // The exporter is invoked even when there are no metrics
+                    if (metricItem.Metrics.Count > 0)
+                    {
+                        metricItems.Add(metricItem);
+                    }
+                }
+            }
+
+            this.meterProvider = Sdk.CreateMeterProviderBuilder()
+                .AddAspNetCoreInstrumentation()
+                .AddMetricProcessor(new PushMetricProcessor(exporter: metricExporter, exportIntervalMs: 10, isDelta: true))
+                .Build();
+
+            using (var client = this.factory.CreateClient())
+            {
+                var response = await client.GetAsync("/api/values");
+                response.EnsureSuccessStatusCode();
+            }
+
+            // Wait for at least two exporter invocations
+            WaitForMetricItems(metricItems, 2);
+
+            this.meterProvider.Dispose();
+
+            // There should be more than one result here since we waited for at least two exporter invocations.
+            // Is this what we should expect?
+            // Since we've configured delta aggregation does it make sense to continue to export zero values?
+            var requestMetrics = metricItems
+                .SelectMany(item => item.Metrics.Where(metric => metric.Name == "http.server.request_count"))
+                .ToArray();
+
+            Assert.True(requestMetrics.Length > 1);
+
+            var metric = requestMetrics[0] as ISumMetric;
+            Assert.NotNull(metric);
+            Assert.Equal(1L, metric.Sum.Value);
+
+            var method = new KeyValuePair<string, object>(SemanticConventions.AttributeHttpMethod, "GET");
+            var scheme = new KeyValuePair<string, object>(SemanticConventions.AttributeHttpScheme, "http");
+            var statusCode = new KeyValuePair<string, object>(SemanticConventions.AttributeHttpStatusCode, 200);
+            var flavor = new KeyValuePair<string, object>(SemanticConventions.AttributeHttpFlavor, "HTTP/1.1");
+            Assert.Contains(method, metric.Attributes);
+            Assert.Contains(scheme, metric.Attributes);
+            Assert.Contains(statusCode, metric.Attributes);
+            Assert.Contains(flavor, metric.Attributes);
+            Assert.Equal(4, metric.Attributes.Length);
+
+            for (var i = 1; i < requestMetrics.Length; ++i)
+            {
+                metric = requestMetrics[i] as ISumMetric;
+                Assert.NotNull(metric);
+                Assert.Equal(0L, metric.Sum.Value);
```

Comment:
why 0 ?

Reply:
oh never mind. Just saw the starting index. This can be revisited, when we modify the SDK to send nulls, instead of 0.